src/LiFi.unit.spec.ts

```diff
@@ -1,28 +1,86 @@
 import { ChainId, CoinKey, findDefaultToken, Token } from '@lifi/types'
-import { beforeEach, describe, expect, it, vi } from 'vitest'
-import { buildStepObject } from '../test/fixtures'
+import { beforeEach, describe, expect, it, Mock, vi } from 'vitest'
+import {
+  buildRouteObject,
+  buildStepObject,
+  mockChainsResponse,
+  mockTransactionRequest,
+} from '../test/fixtures'
 import * as balance from './balance'
 import { convertQuoteToRoute } from './helpers'
 import { LiFi } from './LiFi'
+import { Signer } from 'ethers'
+
+const step = buildStepObject({
+  includingExecution: true,
+})
+
+vi.mock('./services/ApiService', () => ({
+  default: {
+    getStepTransaction: vi.fn(() =>
+      Promise.resolve({ ...step, transactionRequest: mockTransactionRequest })
+    ),
+    getChains: vi.fn(() => Promise.resolve(mockChainsResponse)),
+    getStatus: vi.fn(() =>
+      Promise.resolve({
+        status: 'DONE',
+        receiving: true,
+        sending: {
+          amount: '123',
+          gasAmount: '123',
+          gasAmountUSD: '123',
+          gasPrice: '123',
+          gasToken: '123',
+          gasUsed: '123',
+        },
+      })
+    ),
+  },
+}))
```

Comment:
It looks okay, but can we please use `msw` here, like in the rest of the tests, to mock responses from our BE? The same applies to all tests, if any.

Reply:
Sure, I'll make this change